### PR TITLE
Add Heartbeat and managed annotation checker to the bridge

### PR DESCRIPTION
**Pull request by jaronoff97**
**Description:** 
* Updates to latest opamp-go
* Implements a heartbeat to keep connection alive
* Updates the label check to comply with spec

**Link to tracking Issue:** #2132 

**Testing:** Manual testing and some unit tests... more to come.

### CODE REVIEW

File: cmd/operator-opamp-bridge/operator/client.go [hunk 1]
```diff
@@ -156,14 +178,20 @@ func (c Client) ListInstances() ([]v1alpha1.OpenTelemetryCollector, error) {
 	ctx := context.Background()
 	result := v1alpha1.OpenTelemetryCollectorList{}
 	err := c.k8sClient.List(ctx, &result, client.MatchingLabels{
-		ResourceIdentifierKey: ResourceIdentifierValue,
+		ManagedLabelKey: c.name,
```
Comment (jaronoff97) [inline cmd/operator-opamp-bridge/operator/client.go:181]: i couldn't figure out how to do an `in` query here 🙃

**Review by matej-g — APPROVED** [review]
Looks good to me, nicely done 👍

File: cmd/operator-opamp-bridge/operator/client.go (hunk 1 above)
Comment (swiatekm-sumo) [reply cmd/operator-opamp-bridge/operator/client.go:181]: Like this, I believe:

```go
import (
    "k8s.io/apimachinery/pkg/labels"
    "k8s.io/apimachinery/pkg/selection"
)

labelSelector := labels.NewSelector()
requirement, err := labels.NewRequirement(ManagedLabelKey, selection.In, []string{c.name, "true"})
labelSelector.Add(*requirement)
k8sClient.List(ctx, &result, client.MatchingLabelsSelector{labelSelector})
```

**Review by swiatekm-sumo — APPROVED** [review]
LGTM, left two nitpicks which shouldn't block merging.

**Review by yuriolisa — APPROVED** [review]

**Pull request merged**